numba/np/random/distributions.py

```diff
@@ -529,3 +529,165 @@ def random_poisson(bitgen, lam):
+                continue
+            y = int(xl + np.log(v) / laml)
+            if ((y < 0) or (v == 0.0)):
+                case = 0
+                continue
+            v = v * (u - p2) * laml
+            case = 4
+            continue
+        elif case == 3:
+            y = int(xr - np.log(v) / lamr)
+            if ((y > n) or (v == 0.0)):
+                case = 0
+                continue
+            v = v * (u - p3) * lamr
+            case = 4
+            continue
+        elif case == 4:
+            k = abs(y - m)
+            if ((k > 20) and (k < ((nrq) / 2.0 - 1))):
+                case = 5
+                continue
+            s = r / q
+            a = s * (n + 1)
+            F = 1.0
+            if (m < y):
+                for i in range(m + 1, y + 1):
+                    F = F * (a / i - s)
+            elif (m > y):
+                for i in range(m + 1, y + 1):
+                    F = F / (a / i - s)
+            if (v > F):
+                case = 0
+                continue
+            case = 6
+            continue
+        elif case == 5:
+            rho = (k / (nrq)) * \
+                  ((k * (k / 3.0 + 0.625) + 0.16666666666666666) /
+                   nrq + 0.5)
+            t = -k * k / (2 * nrq)
+            A = np.log(v)
+            if (A < (t - rho)):
+                case = 6
+                continue
+            if (A > (t + rho)):
+                case = 0
+                continue
+            x1 = y + 1
+            f1 = m + 1
+            z = n + 1 - m
+            w = n - y + 1
+            x2 = x1 * x1
+            f2 = f1 * f1
+            z2 = z * z
+            w2 = w * w
+            if (A > (xm * np.log(f1 / x1) + (n - m + 0.5) * np.log(z / w) +
+                     (y - m) * np.log(w * r / (x1 * q)) +
+                     (13680. - (462. - (132. - (99. - 140. / f2) / f2) / f2)
+                      / f2) / f1 / 166320. +
+                     (13680. - (462. - (132. - (99. - 140. / z2) / z2) / z2)
+                      / z2) / z / 166320. +
+                     (13680. - (462. - (132. - (99. - 140. / x2) / x2) / x2)
+                      / x2) / x1 / 166320. +
+                     (13680. - (462. - (132. - (99. - 140. / w2) / w2) / w2)
+                      / w2) / w / 66320.)):
+                case = 0
+                continue
+        elif case == 6:
+            if (p > 0.5):
+                y = n - y
+
+    return y
+
+
+@register_jitable
+def random_binomial_inversion(bitgen, n, p):
+    q = 1.0 - p
+    qn = np.exp(n * np.log(q))
+    _np = n * p
+    bound = min(n, _np + 10.0 * np.sqrt(_np * q + 1))
+
+    X = 0
+    px = qn
+    U = next_double(bitgen)
+    while (U > px):
+        X = X + 1
+        if (X > bound):
+            X = 0
+            px = qn
+            U = next_double(bitgen)
+        else:
+            U -= px
+            px = ((n - X + 1) * p * px) / (X * q)
+
+    return X
+
+
+@register_jitable
+def random_binomial(bitgen, p, n):
```

Comment:
A typo I guess : the order of the arguments "p" and "n" are reversed here in the fn signature, compared to NumPyRandomGeneratorType_binomial which calls random_binomial() like this: random_binomial(inst.bit_generator, n, p)

Reply:
Yes, this seems to be a typo.